front/src/types/User.ts

```diff
@@ -1,81 +1,128 @@
-import { LocalStorageUtils } from '../utils/localStorageUtils.utils';
-import { UserTypes } from './UserType';
+import IonicStorageAccessor from '../services/IonicStorageAccessor';
+import UserType from './UserType';
 
 /**
- * This decorator is used to store the user in the local storage on change
- *
- * @param argument
- * @returns
+ * This decorator is used to store the user in the local storage on change.
  */
 const storeUser = (argument: string) => {
   return (target: any, key: string, descriptor: any) => {
     const originalMethod = descriptor.value;
 
     descriptor.value = function (...args: any[]) {
       this[argument] = args[0];
-      LocalStorageUtils.set('user', this);
+      IonicStorageAccessor.set('user', this);
       return originalMethod.apply(target, args);
     };
 
     return descriptor;
   };
 };
 
-export interface UserInterface {
-  userName: string;
-  profilePicture: string;
-  token: string;
-  type: UserTypes;
-}
+/**
+ * Singleton to use throughout the front to get global data about the user.
+ */
+export default class User {
+  private static instance: User;
 
-export class User {
-  public userName: string;
-  public profilePicture: string;
-  public token: string;
-  public type: UserTypes;
+  private username: string | null;
+  private profilePicture: string | null;
+  private token: string | null;
+  private type: UserType | null;
 
-  constructor(
-    userName: string,
-    profilePicture: string,
-    token: string,
-    type: UserTypes
+  private constructor(
+    username?: string,
+    profilePicture?: string,
+    token?: string,
+    type?: UserType
   ) {
-    this.userName = userName;
-    this.profilePicture = profilePicture;
-    this.token = token;
-    this.type = type;
+    this.username = username ? username : null;
+    this.profilePicture = profilePicture ? profilePicture : null;
+    this.token = token ? token : null;
+    this.type = type ? type : null;
```

Comment:
can be written like this this.username = username ?? null, this means take the value if it's not a falsy value else take null. And maybe as this.username = username as username is defined as an optional params, so string or undefined/null (idk which one)